services/spar/src/Spar/Error.hs

```diff
@@ -87,6 +99,12 @@ waiToServant waierr@(Wai.Error status label _) = ServantErr
   , errHeaders      = []
   }
 
+sparToWaiErrorIO :: Log.Logger -> SparError -> MonadIO m => m (Either ServantErr Wai.Error)
```

Comment:
I've never seen inline constraints like this; I get the idea but it's tougher to read IMO; would you mind moving the constraints out front? Cool to know that it's possible, and in theory it'd be nice to have the constraints near the variable themselves, but it makes it very difficult to see which things are arguments and which are constraints 😄

Reply:
Also I've seen that we very often pass the logger around explicitly; is there a particular reason for doing that rather than using `MonadLogger`?

Reply:
I disagree with you on the RankNType thing, but I'll change the type signature and try to not do avoid it in the future.

No deep philosophical reason to not use `MonadLogger` here, but it would be a bigger change involving the interface of wai-utilities.  I would really like to leave that for future work.

Reply:
I've literally never seen this version of using constraints; so I'd advocate for doing it the other way just because it's how it's done in every single code-base I've ever seen,  😄 

Following patterns enhances readability, even if on a case-by-case basis it seem less readable IMHO